Bjorncs/https state monitor

bjorncs opened the pull request.

bjorncs committed:
Order dependencies on scope

Also change scope of 'annotations' to provided.

bjorncs committed:
Use VespaHttpClientBuilder in service-monitor

hakonhall commented on service-monitor/src/main/java/com/yahoo/vespa/service/health/ApacheHttpClient.java:
```diff
@@ -57,16 +54,23 @@ public long getKeepAliveDuration(HttpResponse response, HttpContext context) {
 
-    ApacheHttpClient(URL url, Duration timeout, Duration keepAlive, ConnectionSocketFactory socketFactory) {
-        this(url, makeCloseableHttpClient(url, timeout, keepAlive, socketFactory));
+    private static HttpClientConnectionManager createConnectionManager(SSLConnectionSocketFactory sslConnectionSocketFactory) {
```
I guess this would be ideal in VespaHttpClientBuilder if other clients wants the same behavior.

bjorncs replied:
I have added a new static factory method that does that. See new commits.

bjorncs committed:
Improve configuration of connection manager

- Add static factory for client builder with BasicHttpClientConnectionManager.
- Simplify implementations of ConnectionManagerFactory by improving its interface.

bjorncs committed:
Use VespaHttpClientBuilder.createWithBasicConnectionManager

bjorncs commented:
@hakonhall PTAL

bjorncs commented:
Build failure is caused by a failing C++ test (messagebus_serviceaddress_test_app).

bjorncs commented:
I will create a new PR based on the http-utils module.
